0036-spsp-pull-payments/0036-spsp-pull-payments.md

```diff
@@ -0,0 +1,140 @@
+---
+title: SPSP Pull Payments
+draft: 1
+---
+# SPSP Pull Payments
+
+## Preface
+
+This document describes how to conduct Pull Payments via [STREAM](../0029-stream/0029-stream.md) once the payment details have been exchanged via the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md) (SPSP).
+
+## Introduction
+
+### Pull Payments
+In contrast to a push payment, where the Payer (sender) actively sends (pushes) units of value to the Payee (receiver), Pull Payments shift the active part from the Payer to the Payee. With the Payer's consent, the Payee can pull units of value directly from the Payer's account to their own account. The only active engagement of the Payer is required during the negotiation phase of the Pull Payment Agreement. 
+
+### Motivation
+
+The [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md) (SPSP) only describes how the details required for a STREAM connection are exchanged and how a simple push payment is made. However, it lacks a detailed explaination of how a Pull Payment is conducted using this connection. 
+
+### Scope
+
+This document specifies how to conduct a Pull Payment once a STREAM connection has been set up. It is intended for use by end-user applications.
+
+### Definitions
+* **Payer** - The entity that units of value are pulled from by the Payee. It is running the [SPSP Server](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions).
+* **Payee** - The entity that is pulling units of value from the Payer. It is running the [SPSP Client](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions).
+* **Pull Payment** - The process of pulling units of value from one account into another account.
+* **Pull Payment Agreement** - An agreement between the Payer and the Payee that specifies how much value can be pulled how often and for how long. 
+* **Token** - An opaque string that represents an identifier for the Pull Payment Agreement. It is part of the [Payment Pointer](../0026-payment-pointers/0026-payment-pointers.md) the SPSP Client needs to connect to the SPSP Server. 
+
+### Operation Overview
+
+The Payee's SPSP Client will set up a [STREAM](../0029-stream/0029-stream.md) connection to the Payer's SPSP Server using the Token, as described by the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md). On connection, the SPSP Client tries to pull as much value from the SPSP server as possible given the Pull Payment Agreement.
+
+## Model of Operation
+
+### Negotiating the Pull Payment Agreement
```

Comment:
_forgot to click add review comment_
I'm on the fence deciding if this should be described as a negotiation. With Codius, the user asks host for the price. The host provides the amount and interval at which they would pull/charge. The user either sets the details and token (because they accept the price) or does nothing (because they don't).
What if this section just describes that the Payer sets the agreement/schedule/rules (with or without input from the Payee) and creates the corresponding token, etc?

Reply:
I don't see a reason why there could not be more of a negotiation. It is just very simple in the Codius use case. In other use cases, it could be something like 
* the merchant suggests an asset 
* the user does not like that and proposes another asset
* the merchant agrees
* the user creates the token
In essence, however, we may just need another spec for the "negotiation" or however we want to call it.